internal/blueprint/filesystem_customizations.go

```diff
@@ -21,20 +22,62 @@ func (fsc *FilesystemCustomization) UnmarshalTOML(data interface{}) error {
 	default:
 		return fmt.Errorf("TOML unmarshal: mountpoint must be string, got %v of type %T", d["mountpoint"], d["mountpoint"])
 	}
+	var size uint64
+	var minsize uint64
 
 	switch d["size"].(type) {
 	case int64:
-		fsc.MinSize = uint64(d["size"].(int64))
+		size = uint64(d["size"].(int64))
 	case string:
-		size, err := common.DataSizeToUint64(d["size"].(string))
+		s, err := common.DataSizeToUint64(d["size"].(string))
 		if err != nil {
 			return fmt.Errorf("TOML unmarshal: size is not valid filesystem size (%w)", err)
 		}
-		fsc.MinSize = size
+		size = s
+	case nil:
+		size = 0
 	default:
 		return fmt.Errorf("TOML unmarshal: size must be integer or string, got %v of type %T", d["size"], d["size"])
 	}
 
+	switch d["minsize"].(type) {
+	case int64:
+		minsize = uint64(d["minsize"].(int64))
+	case string:
+		s, err := common.DataSizeToUint64(d["size"].(string))
+		if err != nil {
+			return fmt.Errorf("TOML unmarshal: size is not valid filesystem size (%w)", err)
+		}
+		minsize = s
+	case nil:
+		minsize = 0
+	default:
+		return fmt.Errorf("TOML unmarshal: minsize must be integer or string, got %v of type %T", d["minsize"], d["minsize"])
+	}
+
+	if size == 0 && minsize == 0 {
+		return fmt.Errorf("TOML unmarshal: minsize must be specified, got %v", minsize)
+	}
+
+	if size > 0 && minsize == 0 {
+		fsc.MinSize = size
+		return nil
+	}
+
+	if size == 0 && minsize > 0 {
+		fsc.MinSize = minsize
+		return nil
+	}
+
+	if size == minsize {
+		fsc.MinSize = size
```

Comment:
nitpick: I would probably take `minsize` here, since it is the preferred option, although the end result is exactly the same.

I know that I suggested this, but wondering what others think - if we should not error in case both options are set to a value. Basically discouraging people to have both set...

Reply:
Yeah I would agree with you, it seems very weird to be able to specify both... so maybe we should discourage this.
I will default to an error and if anyone feels strongly about it we can discuss.

Reply:
I think making it an error is better than silently overriding one or the other.  It can create a lot of confusion.  An error message explaining that both can't be used and that one is an alias for the other is clearer and tells the user  everything they need to know.